internal/pkg/runtime/engine/singularity/prepare_linux.go

```diff
@@ -1440,3 +1446,51 @@ func (e *EngineOperations) loadImage(path string, writable bool) (*image.Image,
 
 	return imgObject, imgErr
 }
+
+func (e *EngineOperations) setUserInfo(useTargetIDs bool) {
+	var gids []int
+
+	pw, err := user.Current()
```

Comment:
Having looked at things... I think that using `user.CurrentOriginal` here, instead, will revert the behaviour to previous w.r.t. the unshare observation. That's what was used previously at `getHomePaths` in container_linux.go